src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -147,6 +149,33 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			this._notebookOptions.layoutChanged(() => this._layoutChanged.fire());
 		}
 		this._defaultKernel = _notebookOptions.defaultKernel;
+		if (this._notebookService) {
+			this._register(this._notebookService.onNotebookKernelsAdded(async kernels => this.handleNewKernelsAdded(kernels).catch(error => onUnexpectedError(error))));
+		}
+	}
+
+	private async handleNewKernelsAdded(kernels: notebookUtils.IStandardKernelWithProvider[]): Promise<void> {
```

Comment:
Kernels still exist when an extension is uninstalled. They get cleared after reloading ADS.

Reply:
That's a bug then, right? If an extension contributing a kernel is uninstalled then we shouldn't be showing/listing that anymore.

Reply:
Kind of. I don't think removing a kernel after it's added is a behavior we have, since it affects currently running sessions and open editors. Like if you uninstall an extension that contributes its own file type with its own deserialization behavior, what happens to those file types that are currently open?

Reply:
Reloading ADS is a common enough behavior after uninstalling an extension. It gives you a button prompt for it right in the extension list.

Reply:
Ok, I looked into this and could swear that there were cases where it let you uninstall an extension without reloading but if an extension is running it just flat out requires a reload. So yeah then just having the event for kernels being added is fine.